src/CustomerProfile.js

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useState } from "react";
+
+export default function CustomerProfile({ id }) {
+  // console.log("CustomerProfile props:", id);
+  const [clientData, setClientData] = useState(null);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
```

Comment:
Just a minor suggestion here, if the variable is a boolean, naming it as `isLoading, setIsLoading, isError ...` is more intuitive.

This reference is of Java but I think this convention is widely adapted 
https://geosoft.no/javastyle.html#Specific